### PR TITLE
Feature/hooks update

### PR DESCRIPTION
**Pull request by anton-karlovskiy**
Sub tasks:
https://github.com/GoogleChromeLabs/adaptive-loading/issues/70#issuecomment-546646634

### CODE REVIEW

File: hooks/README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Enable web developers to differentially deliver lightweight, fast experiences fo
 Now that the adaptive React hooks are [custom hooks](https://reactjs.org/docs/hooks-custom.html) we can just use them like normal React hooks after importing them.
```
Comment (addyosmani) [inline hooks/README.md]: The changes I wanted to review first were to the README - e.g were we able to make it easier to use the Hooks. I think you did a great job here reducing what needs to be imported to exactly the right level for developers to quickly understand these. Nice work :)

I'll review the implementation in more detail through the day.

Comment (anton-karlovskiy) [reply hooks/README.md]: FYI: hooks/README is also updated.

**Comment by addyosmani** [comment]
I've just read through the updates. For the most part, the implementation updates look solid. One change that stood out to me was the memory hook. Is there a reason we are checking for `window === 'undefined'` here in particular? (the other hooks function fine in a client-side context without this particular check).

```const isClientSideRendering = typeof window === 'undefined';
```

**Comment by anton-karlovskiy** [comment]
@addyosmani 
That's just for Next.js animation demo project. Without that conditional, the memory hook is error prone while the component with it is server side rendered. From my experience some node packages are also error prone when they are used in Next.js projects due to the usage of `window` object while server side rendering.

**Comment by addyosmani** [comment]
Ah, I see. For consistency, I wonder if we should be avoiding that check directly in the Hooks and instead update the Next.js animation demo to include the check itself. Wdyt?

**Comment by anton-karlovskiy** [comment]
@addyosmani
At first when I was working on that Next.js demo, I tried that way.
But in my opinion, it might be impossible to call hook conditionally according to rules of hooks.

**Comment by addyosmani** [comment]
I think there are at least two options to move forward here:

1. All hooks have a similar check for consistency so that they can be used in Next.js. I still feel like it might be too early for us to do this as it's not yet clear how much they will be used in that context.

2. We copy the memory hook with the client-side check into the Next.js animation project and use a custom version there but keep the main one in the root hooks directory free of this change. That should allow the animation project to keep working without the check only being present on the memory hook.

I'm leaning towards option 2 (for now) but think in the future we may need to revisit 1. Does that seem fair? Anything I'm missing?

**Comment by anton-karlovskiy** [comment]
@addyosmani 
Yes, I'd like to follow your opinion. option 2 sounds better to me.

**Comment by anton-karlovskiy** [comment]
@addyosmani 

The PR is updated based on option 2.

**Pull request merged**